doc/sphinx/language/core/modules.rst

```diff
@@ -1010,3 +1010,75 @@ subdirectories of path). See the command :cmd:`Declare ML Module` in
 
 See :ref:`command-line-options` for a more general view over the Coq command
 line options.
+
+.. _controlling-locality-of-commands:
+
+Controlling the scope of commands with locality attributes
+----------------------------------------------------------
+
+Many commands have effects that apply only within a specific scope,
+typically the section or the module in which the command was
+called. Locality :term:`attributes <attribute>` can alter the scope of
+the effect. Below, we give the semantics of each locality attribute
+but also alert that a few commands give a different meaning to the
+:attr:`local` and :attr:`global` attributes.
+
+.. attr:: local
+
+   The :attr:`local` attribute limits the effect of the command to the
+   current scope (section or module).
+
+   The ``Local`` prefix is an alternative syntax for the :attr:`local`
+   attribute (see :n:`@legacy_attr`).
+
+   .. note::
+
+      - For some commands, this is the only supported locality within
+        sections (e.g., for :cmd:`Notation`, :cmd:`Ltac` and
+        :ref:`Hint <creating_hints>` commands).
+
+      - For some other commands, this is the default locality within
+        sections even though other locality attributes are supported
+        as well (e.g., it is the case for the :cmd:`Arguments`
+        command).
+
+   .. warning::
+
+      **Exception:** when :attr:`local` is applied to
+      :cmd:`Definition`, :cmd:`Theorem` or their variants, its
+      semantics is different: it makes the defined objects available
+      only through their fully-qualified names rather than their
+      unqualified names after an :cmd:`Import`.
```

Comment:
```suggestion
      :cmd:`Definition`, :cmd:`Theorem` or their variants, its
      semantics are different: when these objects are :cmd:`Import`\ed,
      they must be referenced with their fully-qualified names rather than their
      unqualified names.
```

Reply:
I've applied the "is" -> "are" change but not the rest because I don't want to lose the idea of the effect of `local` that is carried by "it makes".